hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -1520,8 +1518,10 @@ protected void stopServiceThreads() {
 
   private void createProcedureExecutor() throws IOException {
     MasterProcedureEnv procEnv = new MasterProcedureEnv(this);
-    procedureStore =
-      new RegionProcedureStore(this, new MasterProcedureEnv.FsUtilsLeaseRecovery(this));
+    // Create cleaner thread pool
+    cleanerPool = new DirScanPool(conf);
```

Comment:
Strange place to start cleanerPool down here in the procedure startup given it is used cleaning WALs and hfile...

Reply:
But we have no choice as we need to use it in the next line...

Reply:
I was thinking you'd call it up in the caller's method, in startServiceThreads, rather than down hidden in here in startProcedureExecutor... i.e. move the line up thee lines from where it was.

Reply:
The startServiceThreads method is called after createProcedureExecutor, notice that, it is create, not start...